### ISSUE TITLE
[Bug]: Accessibility issue with Fluent UI Dialog on Android, iOS and Mac Devices

### ISSUE THREAD

**Issue by hackvimal** [microsoft/fluentui#27806]
### Library

React v8 (@fluentui/react)

### System Info

```shell
Android, iOS and mac
```

### Are you reporting Accessibility issue?

yes

### Reproduction

https://codepen.io/Abhijeet-Vimal/pen/QWZmRWJ

### Bug Description

Description:
When opening a Fluent UI dialog on Android, iOS, and Mac devices, the TalkBack and VoiceOver screen readers do not announce the title and subtitle of the dialog, but only the button. This behavior was also verified on the Fluent UI sample dialog.

Steps to reproduce:

Open any Fluent UI dialog on an Android, iOS, or Mac device.
Use the TalkBack or VoiceOver screen reader to navigate the dialog.
Observe that the screen reader does not announce the dialog title and subtitle, but only the button.

Expected result:
The screen reader should announce both the title and subtitle of the Fluent UI dialog in addition to the button when the dialog is opened.

Actual result:
The screen reader only announces the button when opening a Fluent UI dialog on Android, iOS, and Mac devices.

Reproducible:
Always

Link to codepen - https://codepen.io/Abhijeet-Vimal/pen/QWZmRWJ

### Requested priority

High

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.
